actionpack/lib/action_controller/metal/rendering.rb

```diff
@@ -4,6 +4,17 @@ module Rendering
 
     RENDER_FORMATS_IN_PRIORITY = [:body, :text, :plain, :html]
 
+    module ClassMethods
+      # Documentation at ActionController::Renderer#render
+      delegate :render, to: :renderer
+
+      # Returns a renderer class (inherited from ActionController::Renderer)
+      # for the controller.
+      def renderer
+        @renderer ||= Renderer.for(self)
```

Comment:
Yes, this will return a class to create a new instances of renderer from. And even if you use one instance of renderer across threads it's still be ok, since actual rendering happens always on a new controller instance.